src/app/forms/components/suggestive-input/suggestive-input.component.ts

```diff
@@ -0,0 +1,78 @@
+import { AfterContentInit, ChangeDetectionStrategy, ChangeDetectorRef, Component, Injector, Input, OnInit } from '@angular/core';
+import { NG_VALUE_ACCESSOR } from '@angular/forms';
+import { MultiOption } from '@forms/models/options.model';
+import { CustomValidators } from '@forms/validators/custom-validators';
+import { firstValueFrom, Observable, skipWhile, take } from 'rxjs';
+import { BaseControlComponent } from '../base-control/base-control.component';
+
+@Component({
+  selector: 'app-suggestive-input',
+  templateUrl: './suggestive-input.component.html',
+  styleUrls: ['./suggestive-input.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: SuggestiveInputComponent,
+      multi: true
+    }
+  ]
+})
+export class SuggestiveInputComponent extends BaseControlComponent implements AfterContentInit, OnInit {
+  @Input() options$!: Observable<MultiOption[]>;
+  @Input() defaultValue: string = '';
+
+  _value = '';
+
+  constructor(injector: Injector, changeDetectorRef: ChangeDetectorRef) {
+    super(injector, changeDetectorRef);
+  }
+
+  ngOnInit(): void {
+    this.options$
+      .pipe(
+        skipWhile(options => !options.length),
+        take(1)
+      )
+      .subscribe(async () => {
+        this._value = (await this.findOption(this.value, 'value'))?.label ?? '';
+      });
+  }
+
+  override async ngAfterContentInit(): Promise<void> {
+    super.ngAfterContentInit();
+    this.addValidators();
+  }
+
+  get style(): string {
+    return 'govuk-input ' + (this.width ? 'govuk-input--width-' + this.width : '');
+  }
+
+  async handleChangeForOption(value: string) {
+    const option = await this.findOption(value);
+
+    this._value = option?.label ?? value;
+    // if value, patch option value else `[INVALID_OPTION]` if not found, finally propagate empty value
+    this.control?.patchValue(option ? option.value : value ? '[INVALID_OPTION]' : value);
```

Comment:
Is the last check correct here?

If `option` then use `option.value`, otherwise if `value` use `'[INVALID_OPTION]'`?

Reply:
That makes sense, cheers 👍